### PR TITLE
Verify and fix code format automatically by Spotless

### PR DESCRIPTION
**Pull request by KengoTODA**
By this PR, we verify code automatically in CI. For instance, [this build](https://travis-ci.org/spotbugs/spotbugs-gradle-plugin/builds/494690094) was failed because it has wrongly formatted code. Then reviewer can focus not on code format but on important things.

Spotless also provides automatic fix, we can run `./gradlew spotlessApply` then it will fix all problems in the project.

### CODE REVIEW

File: gradle/spotless.gradle [hunk 1]
```diff
@@ -0,0 +1,7 @@
+spotless {
+  java {
+    removeUnusedImports()
```
Comment (wreulicke) [inline gradle/spotless.gradle]: why do not use any format such as eclipseFormatter or googleJavaFormat?

https://github.com/diffplug/spotless/blob/master/plugin-gradle/README.md#applying-to-java-source-with-google-java-format

**Comment by wreulicke** [comment]
This is just a question.

Does this PR introduce format lint with spotless on travisci?

**Comment by KengoTODA** [comment]
> Does this PR introduce format lint with spotless on travisci?

`spotlessCheck` task depends on `check` task, so it will run in the CI build:
https://github.com/spotbugs/spotbugs-gradle-plugin/blob/84b55ebfd723bdfb6898be5cacc372499b8ff64b/.travis.yml#L42

However `spotlessApply` does not run in CI for now.

File: gradle/spotless.gradle (hunk 1 above)
Comment (KengoTODA) [reply gradle/spotless.gradle]: It's also possible solution, I think. I just wanted to keep the diff simple.

@spotbugs/everyone if you don't care, I will introduce the latest googleJavaFormat.

**Comment by ThrawnCA** [comment]
Is this going to reformat the whole codebase at once?

Won't that mean a huge whitespace change in the commit history?

**Comment by KengoTODA** [comment]
@ThrawnCA if we apply google java format, it is possible. Current change set doesn't contain big change, because we apply minimal format rules only.

**Comment by KengoTODA** [comment]
Hello, can we merge this PR? Then please approve.

I think we don't need to add more rules in this timing.

**Pull request merged**